File: onyx-database-tests/src/main/java/com/onyxdevtools/server/entities/Actor.java
package com.onyxdevtools.server.entities;

import com.onyx.persistence.IManagedEntity;
import com.onyx.persistence.annotations.*;
import com.onyx.persistence.annotations.values.CascadePolicy;
import com.onyx.persistence.annotations.values.RelationshipType;

@Entity
public class Actor extends Person implements IManagedEntity
{

    @Identifier
    @Attribute
    public int actorId;

    @Attribute
    public String firstName;

    @Attribute
    public String lastName;

    @Relationship(type = RelationshipType.MANY_TO_ONE,
            cascadePolicy = CascadePolicy.NONE,
            inverseClass = Movie.class,
            inverse = "actors")
    public Movie movie;
}

File: onyx-database-tests/src/main/kotlin/com/onyxdevtools/server/entities/Actor.kt
package dev.onyx.server.entities

import com.onyx.persistence.IManagedEntity
import com.onyx.persistence.annotations.*
import com.onyx.persistence.annotations.values.CascadePolicy
import com.onyx.persistence.annotations.values.RelationshipType

@Entity
class Actor : Person(), IManagedEntity {

    @Identifier
    var actorId: Int = 0

    @Relationship(type = RelationshipType.MANY_TO_ONE, cascadePolicy = CascadePolicy.NONE, inverseClass = Movie::class, inverse = "actors")
    @Suppress("unused")
    var movie: Movie? = null
}
